core/services/relay/evm/event_binding.go

```diff
@@ -130,16 +105,31 @@ func (e *eventBinding) GetLatestValue(ctx context.Context, params, into any) err
 }
 
 func (e *eventBinding) QueryKey(ctx context.Context, filter query.KeyFilter, limitAndSort query.LimitAndSort, sequenceDataType any) ([]commontypes.Sequence, error) {
-	remappedFilter, err := e.remapFilter(filter)
+	if !e.bound {
+		return nil, fmt.Errorf("%w: event not bound", commontypes.ErrInvalidType)
+	}
+
+	remapped, err := e.remap(filter)
 	if err != nil {
 		return nil, err
 	}
 
-	logs, err := e.lp.FilteredLogs(remappedFilter, limitAndSort)
+	// filter should always use the address and event sig
+	defaultExpressions := []query.Expression{
+		logpoller.NewAddressFilter(e.address),
+		logpoller.NewEventSigFilter(e.hash),
+	}
+	remapped.Expressions = append(defaultExpressions, remapped.Expressions...)
+
+	logs, err := e.lp.FilteredLogs(ctx, remapped, limitAndSort)
 	if err != nil {
 		return nil, err
 	}
 
+	if len(logs) == 0 {
+		return nil, commontypes.ErrNotFound
+	}
+
```

Comment:
I don't think that we should treat this as an error, SelectLogsWithSigs does this:
```
if pkgerrors.Is(err, sql.ErrNoRows) {
		return nil, nil
	}
```
Other LogPoller querying methods just return empty slice of logs. Probably makes sense to just return an empty slice of sequences?

Reply:
Works for me

Reply:
I found out why I originally returned an error here. The interface tests expect a not found error: `assert.True(t, errors.Is(err, types.ErrNotFound))`

Reply:
Which test, how did log poller pass that test before?

Reply:
`RunQueryKeyInterfaceTests`: the new tests for chain reader

Reply:
Pending PR: https://github.com/smartcontractkit/chainlink-common/pull/508

This PR alters the tests to not return an error, but instead and empty slice.

Reply:
Approved, then should this also return an empty slice?

Reply:
Correct